packages/melos/lib/src/common/io.dart

```diff
@@ -0,0 +1,288 @@
+/// logged.
+void writeTextFile(
+  String file,
+  String contents, {
+  bool dontLogContents = false,
+  Encoding encoding = utf8,
+  bool recursive = false,
+}) {
+  deleteIfLink(file);
+  final fileObject = File(file);
+  if (recursive) {
+    fileObject.createSync(recursive: true);
+  }
+  File(file).writeAsStringSync(contents, encoding: encoding);
+}
+
+/// Creates [file] and writes [contents] to it.
+///
+/// If [dontLogContents] is `true`, the contents of the file will never be
+/// logged.
+Future<void> writeTextFileAsync(
+  String file,
+  String contents, {
+  bool dontLogContents = false,
+  Encoding encoding = utf8,
+  bool recursive = false,
+}) async {
+  deleteIfLink(file);
+  final fileObject = File(file);
+  if (recursive) {
+    await fileObject.create(recursive: true);
+  }
+  await fileObject.writeAsString(contents, encoding: encoding);
+}
+
+/// Ensures that [dir] and all its parent directories exist.
+///
+/// If they don't exist, creates them.
+String ensureDir(String dir) {
+  Directory(dir).createSync(recursive: true);
+  return dir;
+}
+
+/// Creates a temp directory in [base], whose name will be [prefix] with
+/// characters appended to it to make a unique name.
+///
+/// Returns the path of the created directory.
+String createTempDir(String base, [String? prefix]) =>
+    Directory(base).createTempSync(prefix).path;
+
+/// Copies a file [from] the source file [to] the destination file.
+void copyFile(String from, String to, {bool recursive = false}) {
+  if (recursive) {
+    File(to).createSync(recursive: true);
+  }
+  File(from).copySync(to);
+}
+
+/// Deletes [file] if it's a symlink.
+///
+/// The [File] class overwrites the symlink targets when writing to a file,
+/// which is never what we want, so this delete the symlink first if necessary.
+void deleteIfLink(String file) {
+  if (!linkExists(file)) return;
+  Link(file).deleteSync();
+}
+
+/// Deletes whatever's at [path], whether it's a file, directory, or symlink.
+///
+/// If it's a directory, it will be deleted recursively.
+void deleteEntry(String path) {
+  _attempt('delete entry', () {
+    if (linkExists(path)) {
+      Link(path).deleteSync();
+    } else if (dirExists(path)) {
+      Directory(path).deleteSync(recursive: true);
+    } else if (fileExists(path)) {
+      File(path).deleteSync();
+    }
+  });
+}
+
+extension FileSystemEntityUtils on FileSystemEntity {
+  /// Tries to resolve the path of this [FileSystemEntity] through
+  /// [resolveSymbolicLinks] and returns `null` if the path cannot be resolved.
+  ///
+  /// For example, a path cannot be resolved when it is a link to a non-existing
+  /// file.
+  Future<String?> tryResolveSymbolicLinks() async {
+    try {
+      return await resolveSymbolicLinks();
+    } on FileSystemException {
+      return null;
+    }
+  }
+}
+
+extension DirectoryUtils on Directory {
+  /// Lists the sub-directories and files of this [Directory] similar to [list].
```

Comment:
```suggestion
  /// Lists the sub-directories and files of this [Directory] similar to [list].
  ///
```

Nit: [Separate the first sentence of a doc comment into its own paragraph.](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph)

Reply:
Thanks :)